last-order-api/src/main/java/com/flab/favorite/service/FavoriteService.java

```diff
@@ -0,0 +1,55 @@
+package com.flab.favorite.service;
+
+import com.flab.favorite.domain.Favorite;
+import com.flab.favorite.dto.SearchFavoriteResponse;
+import com.flab.favorite.exception.AlreadyFavoriteException;
+import com.flab.favorite.exception.FavoriteNotExistException;
+import com.flab.favorite.repository.FavoriteRepository;
+import com.flab.store.StoreRepository;
+import com.flab.store.domain.Store;
+import com.flab.store.exception.StoreNotExistException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class FavoriteService {
+
+    private final FavoriteRepository favoriteRepository;
+
+    private final StoreRepository storeRepository;
+
+    public Long addFavorite(Long storeId) {
+        checkStore(storeId);
+        if(favoriteRepository.existsByUserIdAndStoreId(1L, storeId)){
+            throw new AlreadyFavoriteException();
+        }
+        Favorite favorite = Favorite.builder()
+                .userId(1L)
+                .storeId(storeId)
+                .build();
+
+        return favoriteRepository.save(favorite).getId();
+    }
+
+    public void deleteFavorite(Long favoriteId) {
+        Favorite favorite = favoriteRepository.findById(favoriteId).orElseThrow(() -> new FavoriteNotExistException());
+        favoriteRepository.delete(favorite);
+    }
+
+    public List<SearchFavoriteResponse> searchAllMyFavorite(){
+        List<SearchFavoriteResponse> favoriteResponseList = favoriteRepository.findAllByUserId(1L).stream().map(SearchFavoriteResponse::new).collect(Collectors.toList());
+        for (SearchFavoriteResponse searchFavoriteResponse : favoriteResponseList) {
+            searchFavoriteResponse.setFavoriteInfo(checkStore(searchFavoriteResponse.getStoreId()));
+        }
+        return favoriteResponseList;
+    }
+
+    private Store checkStore(Long storeId) {
+        return storeRepository.findById(storeId).orElseThrow(() -> new StoreNotExistException());
+    }
```

Comment:
이 메소드는 크게 두 가지 역할을 하는데, 하나는 storeId가 제대로 되어 있는지 확인하는 것이고, 다른 하나는 Store entity를 리턴하는 것이예요. 그래서 이름은 checkStore인데, 실제로는 Store를 리턴하게 구현이 된 것이죠. 이럴 때는

```suggestion
    private Optional<Store> getStore(Long storeId) {
        return storeRepository.findById(storeId);
    }
```

이렇게 하고, 이 메소드를 부르는 쪽에서 저 메소드에 값이 정확히 들어있는지 스스로 확인하게 만드는게 더 명확한 코드 일 것 같습니다.